baybe/acquisition/base.py

```diff
@@ -25,13 +29,44 @@ class AcquisitionFunction(ABC, SerialMixin):
     _abbreviation: ClassVar[str]
     """An alternative name for type resolution."""
 
-    def to_botorch(self, surrogate: Surrogate, best_f: float):
+    @classmethod
+    @property
```

Comment:
seems this chaining is deprecate in Python
https://github.com/python/mypy/issues/13746

Reply:
@AdrianSosic do you know any fancy schmancy thing to get around this? I thought about making this a non-classmethod property but to me a classmethod+property is really what I want. But looking online there really doesnt seem any python way of doing it

Reply:
Hm, the only other alternatives that currently come to mind are
* using explicit class variables, but of course inelegant since needs to be manually maintained ...
* using a metaclass to accomplish the same, but that's typically a rabbit hole you don't want to enter

I also often feel that true "class properties" are a gap in python .... and now that they have deprecated the stacking, I wonder what is the "recommended" alternative to it!? It's interesting that all the discussions seem to center around deprecating it and mentioning problems, but not pointing out alternatives!? 🤔 

How deep did you go through the threads? Is it worth just asking what they suggest to use instead?

Reply:
Well, it should be a property, and it should be a classmethod (conceptually) :D

Reply:
@Scienfitz: for the sake of completeness

```python
from attrs import define


class MCMeta(type):
    def __new__(cls, name, bases, dct):
        x = super().__new__(cls, name, bases, dct)
        x.is_mc = name.startswith("q")
        return x

def add_mc(cls):
    cls.is_mc = cls.__name__.startswith("q")
    return cls


@define
class qEI(metaclass=MCMeta):
    pass

@add_mc
class qEI():
    pass
```

Reply:
And finally (and this is probably the "correct" way, because other decorators like `property` and `classmethod` also work in that way, is to use descriptors:

```python
class classproperty:
    def __init__(self, fn) -> None:
        self.fn = fn

    def __get__(self, _, cls):
        return self.fn(cls)

class qEI:
    @classproperty
    def is_mc(cls):
        return cls.__name__.startswith("q")
```

I guess this is what you want?

Reply:
perfect and many thx, I added this to utils
seems you could get a lot fo upvotes for that on stackoverflow, doesnt seem many good solutions out there

Reply:
You're welcome. Haven't checked the `mypyability` though. Is it happy? Autocomplete should work, I think ...

Reply:
(and type hint missing in my `is_mc` function)

Reply:
do you want me to type hiny `cls` or what do you mean? 
`def is_mc(cls) -> bool:`

Reply:
no, the bool

Reply:
ah i see sry got confused between the slight differences in the orig and your example `is_mc`, problem seems solved will resolve